ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -154,18 +151,13 @@ let update_vm_virtual_hardware_platform_version ~__context ~vm =
 
 let create_from_record_without_checking_licence_feature_for_vendor_device
```

Comment:
This whole function no longer serves a purpose now. We should remove it, and have its callers call `Client.VM.create_from_record` directly.

Reply:
What about this part?
```
update_vm_virtual_hardware_platform_version ~__context ~vm ;
```

Reply:
That looks redundant, as `VM.create` does the same. The `update_vm_virtual_hardware_platform_version ` function depends on the current `has_vendor_device` value of the VM, which is why the function-with-the-long-name had call it again after setting the field.